File: VB/ChartsDemo.Wpf/ViewModels/Bar3DViewModel.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Windows.Media.Media3D

Namespace ChartsDemo
    Public Class Bar3DViewModel
        Private Const pointsCount As Integer = 300
        Private Const maxValue As Integer = 200
        Public Overridable Property Source() As List(Of Point3D)

        Public Sub New()
            GenerateData()
        End Sub

        Public Sub GenerateData()
            Source = GenerateRandomData()
        End Sub

        Private Shared Function GenerateRandomData() As List(Of Point3D)
            Dim rand = New Random()
            Dim data = New List(Of Point3D)()
            For i As Integer = 0 To pointsCount - 1
                Dim point As Point3D
                Do
                    Dim x As Double = rand.NextDouble() * maxValue * 2
                    Dim y As Double = rand.NextDouble() * maxValue * 2
                    Dim z As Double = rand.NextDouble() * maxValue
                    point = New Point3D(x, y, z)
                Loop While data.Contains(point, New Point3DComparer(1.0))
                data.Add(point)
            Next i
            Return data
        End Function
    End Class
End Namespace

File: CS/ChartsDemo.Wpf/ViewModels/Bar3DViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Media.Media3D;

namespace ChartsDemo {
    public class Bar3DViewModel {
        const int pointsCount = 300;
        const int maxValue = 200;
        public virtual List<Point3D> Source { get; set; }

        public Bar3DViewModel() {
            GenerateData();
        }

        public void GenerateData() {
            Source = GenerateRandomData();
        }

        static List<Point3D> GenerateRandomData() {
            var rand = new Random();
            var data = new List<Point3D>();
            for(int i = 0; i < pointsCount; i++) {
                Point3D point;
                do {
                    double x = rand.NextDouble() * maxValue * 2;
                    double y = rand.NextDouble() * maxValue * 2;
                    double z = rand.NextDouble() * maxValue;
                    point = new Point3D(x, y, z);
                } while(data.Contains(point, new Point3DComparer(1.0)));
                data.Add(point);
            }
            return data;
        }
    }
}
